fml/shared_thread_merger.cc

```diff
@@ -0,0 +1,88 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#define FML_USED_ON_EMBEDDER
+
+#include "flutter/fml/shared_thread_merger.h"
+
+#include <set>
+
+namespace fml {
+
+SharedThreadMerger::SharedThreadMerger(fml::TaskQueueId owner,
+                                       fml::TaskQueueId subsumed)
+    : owner_(owner),
+      subsumed_(subsumed),
+      task_queues_(fml::MessageLoopTaskQueues::GetInstance()) {}
+
+bool SharedThreadMerger::MergeWithLease(RasterThreadMergerId caller,
+                                        int lease_term) {
+  FML_DCHECK(lease_term > 0) << "lease_term should be positive.";
+  std::scoped_lock lock(mutex_);
+  if (IsMergedUnSafe()) {
+    return true;
+  }
+  bool success = task_queues_->Merge(owner_, subsumed_);
+  FML_CHECK(success) << "Unable to merge the raster and platform threads.";
+  // Save the lease term
+  lease_term_by_caller_[caller] = lease_term;
```

Comment:
Ahh, this is how you solved the double decrement issue, right?

Reply:
@gaaclarke Yes! 😄